api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -96,10 +99,28 @@ message DownstreamTlsContext {
 }
 
 // TLS context shared by both client and server TLS contexts.
-// [#next-free-field: 9]
+// [#next-free-field: 11]
 message CommonTlsContext {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.auth.CommonTlsContext";
 
+  // Config for Certificate provider to get certificates. This provider should allow certificates to be
+  // fetched/refreshed over the network "asynchronously" with respect to the TLS handshake.
+  message CertificateProvider {
+    // opaque name used to specify certificate instances or types. For example, "ROOTCA" to specify
+    // a root-certificate (validation context) or "TLS" to specify a new tls-certificate.
+    string name = 1;
+
+    // Provider specific config.
+    // Note: an implementation is expected to dedup multiple instances of the same config
+    // to maintain a single certificate-provider instance. The sharing can happen, for
+    // example, among multiple clusters or between the tls_certificate and validation_context
+    // certificate providers of a cluster.
+    oneof config {
```

Comment:
Can you add a required validation to the oneof?

Reply:
done